### PR TITLE
Remove IotaDID->DID Deref anti-pattern.

### PR DESCRIPTION
**Pull request by m-renaud**
# Description of change

Remove `Deref` impl from `IotaDID` to `DID`, re-implement methods on `IotaDID`.

## Links to any relevant issues

N/A

## Type of change

Internal cleanup.

## How the change has been tested

`cargo test`

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: identity-iota/src/did/url/iota_did.rs [hunk 1]
```diff
@@ -130,6 +129,16 @@ impl IotaDID {
     self.0.join(other).map_err(Into::into).and_then(Self::try_from_owned)
   }
 
+  /// Change the method of the [`IotaDID`].
+  pub fn set_method(&mut self, value: impl AsRef<str>) {
```
Comment (l1h3r) [inline identity-iota/src/did/url/iota_did.rs:133]: This method can be left out, it should always be `iota`.

File: identity-iota/Cargo.toml [hunk 2]
```diff
@@ -12,6 +12,7 @@ homepage = "https://www.iota.org"
 
 [dependencies]
 async-trait = { version = "0.1", default-features = false }
+form_urlencoded = { version = "1.0.1" }
```
Comment (l1h3r) [inline identity-iota/Cargo.toml:15]: Just specifying `1.0` should be enough; this will help prevent conflicting versions

Comment (m-renaud) [reply identity-iota/Cargo.toml]: Done.

File: identity-iota/src/did/url/iota_did.rs (hunk 1 above)
Comment (m-renaud) [reply identity-iota/src/did/url/iota_did.rs]: Done.

**Review by l1h3r — APPROVED** [review]

**Pull request merged**